### PR TITLE
[MM-54830] Convert LocalizedInput of 'team_general_tab.tsx' to regular input component

### PR DESCRIPTION
**Pull request by akbarkz**
#### Summary

Replace `LocalizedInput` component in `webapp/channels/src/components/team_general_tab/team_general_tab.tsx` to a regular input.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24819
Jira https://mattermost.atlassian.net/browse/MM-54830

#### Screenshots

Screenshot 2023-10-19 at 12 37 29

#### Release Note
```
NONE
```

### CODE REVIEW

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
The changes looks great can you please check the failing CI

**Comment by akbarkz** [comment]
> The changes looks great can you please check the failing CI

Yep, was already fixing it. Sorry, I should have run the checks locally before pushing.

File: webapp/channels/src/components/team_settings/team_settings.tsx [hunk 1]
```diff
@@ -36,6 +39,7 @@ const TeamSettings = ({
                 <GeneralTab
                     team={team}
                     activeSection={activeSection}
+                    intl={intl}
```
Comment (larkox) [inline webapp/channels/src/components/team_settings/team_settings.tsx:42]: Why is this needed? Shouldn't injectIntl inject it itself without us passing it?

File: webapp/channels/src/components/team_general_tab/index.ts [hunk 2]
```diff
@@ -24,6 +25,7 @@ export type OwnProps = {
     updateSection: (section: string) => void;
     team?: Team & { last_team_icon_update?: number };
     activeSection: string;
+    intl: IntlShape;
```
Comment (larkox) [inline webapp/channels/src/components/team_general_tab/index.ts:28]: This should not be needed at this level.

File: webapp/channels/src/components/team_general_tab/team_general_tab.tsx [hunk 3]
```diff
@@ -39,7 +37,7 @@ type State = {
     shouldFetchTeam?: boolean;
 }
 
-export default class GeneralTab extends React.PureComponent<Props, State> {
+export class GeneralTab extends React.PureComponent<Props, State> {
```
Comment (larkox) [inline webapp/channels/src/components/team_general_tab/team_general_tab.tsx:40]: I think the most proper typing for the props in this file would be the following

```
type Props = PropsFromRedux & OwnProps & WrappedComponentProps;
```

`WrappedComponentProps` is exported by `react-intl` and it is that, the props that the wrapped component receives from `injectIntl`.

**Review by larkox — CHANGES_REQUESTED** [review]
Thank you for the contribution! A few changes here and there.

File: webapp/channels/src/components/team_general_tab/team_general_tab.tsx (hunk 3 above)
Comment (akbarkz) [reply webapp/channels/src/components/team_general_tab/team_general_tab.tsx:40]: oh wow, this is brilliant! Indeed adding `react-intl`'s own type to Props solves all the issues. I don't need to add additional property to TeamSettings anymore. Thanks @larkox!

**Review by larkox — APPROVED** [review]
Now it looks great to me. Thanks!

I expect still some lint errors, but let's the CI judge that :P

**Comment by akbarkz** [comment]
> I expect still some lint errors, but let's the CI judge that :P

The CI failed, but apparently not for the linting issues, but some tests failed with timeout exceeded error. I wonder how can I re-run the job?

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by yasserfaraazkhan** [comment]
@M-ZubairAhmed 
wanted to check if this UI change is  intended, didn't see it in other pr ?
image

**Review by yasserfaraazkhan — APPROVED** [review]
